src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+
+                setFields(returnedBook, resultSet.getObject("id", Long.class),
+                        resultSet.getObject("title", String.class),
+                        resultSet.getObject("price", BigDecimal.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed get book by id command.Book "
+                    + "with id" + id, e);
+        }
+
+        return Optional.of(returnedBook);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sqlFindAll = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.connect();
+                PreparedStatement statement = connection.prepareStatement(sqlFindAll)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = new Book();
+
+                setFields(book, resultSet.getObject("id",Long.class),
+                        resultSet.getString("title"),
+                        resultSet.getBigDecimal("price"));
+
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed select all book from db", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sqlUpdateByName = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.connect();
+                PreparedStatement statement = connection.prepareStatement(sqlUpdateByName)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int rowsUpdated = statement.executeUpdate();
+
+            if (rowsUpdated > 0) {
+                return book;
+            } else {
+                throw new DataProcessingException("Failed to update the"
+                        + " book with ID: " + book.getId());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Operation updade failed with book title: "
+                    + book.getTitle(), e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sqlDelete = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.connect();
+                 PreparedStatement statement = connection.prepareStatement(sqlDelete)) {
+            statement.setLong(1, id);
+            int rowsDeleted = statement.executeUpdate();
+            return rowsDeleted > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Delete book failed: id = " + id, e);
+        }
+    }
+
+    private static void setFields(Book returnedBook, Long bookId,
```

Comment:
It's better to modify the logic of this method.
Provide ResultSet as argument and do the logic. 
Also return Book 
Rename the method - mapResultToBook for example